File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.rs
use std::vec::Vec;

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut index1 = 0;
        let mut index2 = numbers.len() - 1;
        while index1 < index2 {
            if numbers[index1] + numbers[index2] > target {
                index2 -= 1;
            } else if numbers[index1] + numbers[index2] == target {
                break;
            } else {
                index1 += 1;
            }
        }
        
        return vec![(index1 + 1) as i32, (index2 + 1) as i32];
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0,
            r = numbers.size() - 1;
        
        while (l < r) {
            int m = numbers[l] + numbers[r];
            
            if (m > target) {
                --r;
            } else if (m < target) {
                ++l;
            } else {
                return {l + 1,r + 1};
            }
        }
        
        return {-1,-1};
    }
};